frontend/cypress/support/commands.ts

```diff
@@ -41,8 +41,211 @@
+                            participantId,
+                            '',
+                            (response: SocketResponse<UUID>) =>
+                                resolve(response)
+                        );
+                    }),
+                    { log: false }
+                )
+                    .its('payload')
+                    .as('participantSocketUUID');
+            }
+        );
+    });
+
+    cy.wrap([])
+        .as('participantSocketPerformedActions')
+        .then((performedActions: any) => {
+            cy.get('@participantSocket', { log: false }).then(
+                (participantSocket: any) => {
+                    participantSocket.on(
+                        'performAction',
+                        (action: ExerciseAction) => {
+                            performedActions.push(action);
+                        }
+                    );
+                }
+            );
+        });
+
+    return cy;
+}
+
+export function initializeTrainerSocket() {
+    cy.get('@websocketBaseUrl', { log: false }).then(
+        (websocketBaseUrl: any) => {
+            cy.wrap(io(websocketBaseUrl, { transports: ['websocket'] })).as(
+                'trainerSocket'
+            );
+        }
+    );
+
+    cy.get('@trainerId', { log: false }).then((trainerId: any) => {
+        cy.get('@trainerSocket', { log: false }).then((trainerSocket: any) => {
+            cy.wrap(
+                new Promise<SocketResponse<UUID>>((resolve) => {
+                    trainerSocket.emit(
+                        'joinExercise',
+                        trainerId,
+                        '',
+                        (response: SocketResponse<UUID>) => resolve(response)
+                    );
+                }),
+                { log: false }
+            )
+                .its('payload')
+                .as('trainerSocketUUID');
+        });
+    });
+
+    cy.wrap([])
+        .as('trainerSocketPerformedActions')
+        .then((performedActions: any) => {
+            cy.get('@trainerSocket', { log: false }).then(
+                (trainerSocket: any) => {
+                    trainerSocket.on(
+                        'performAction',
+                        (action: ExerciseAction) => {
+                            performedActions.push(action);
+                        }
+                    );
+                }
+            );
+        });
+
+    return cy;
+}
+
+export function firstElement(subject: Array<unknown>) {
+    return cy.log('first element').wrap(subject.at(0), { log: false });
+}
+
+export function lastElement(subject: Array<unknown>) {
+    return cy.log('last element').wrap(subject.at(-1), { log: false });
+}
+
+export function atPosition(subject: Array<unknown>, n: number) {
+    if (n < 0) {
+        return cy
+            .log(`${n}th element`)
+            .wrap(subject.at(subject.length + n), { log: false });
+    }
+    return cy.log(`${n}th element`).wrap(subject.at(n), { log: false });
```

Comment:
```suggestion
    return cy.log(`${n}th element`).wrap(subject.at(n), { log: false });
```

This should be equivalent code.

Reply:
Fixed in b501980e368668c216da57909d2ea74c919bebe8

Reply:
Edit: Fixed in 2e505aa4abb4a0ec3d9c92e0ddd60eff102818d3